docs/how-to/modify-kernel-boot-parameters.rst

```diff
@@ -0,0 +1,140 @@
+How to modify kernel boot parameters
+=======================================
+
+The real-time performance-tuning features compiled into the kernel can be
+controlled by various means. The approach used here is a common one:
+passing command line parameters via a bootloader.
+
+The Linux kernel supports a wide range of parameters, but only a few of them are relevant to real-time tuning.
+:doc:`/reference/kernel-boot-parameters` is good a reference.
+
+Depending on the underlying boot mechanism, there are multiple ways to pass command line parameters to the kernel.
+
+For Raspberry Pi, jump to :ref:`raspberry-pi`.
+
+Ubuntu Core uses GRUB but provides a different interface for 
+`modifying kernel options <https://ubuntu.com/core/docs/modify-kernel-options>`_.
+
+.. Replace the Ubuntu Core link to the internal how-to (https://documentation.ubuntu.com/real-time/en/ubuntu-core-devel/how-to/uc-boot-parameters/), once released.
+
+Continue reading for systems that use the GNU GRUB bootloader.
+
+Modify kernel boot parameters via GRUB
+--------------------------------------
+
+You can temporarily or permanently configure parameters which GRUB will pass to
+Real-time Ubuntu.
+
+When first testing a real-time tuning feature, `configure it temporarily
+<#configure-grub-temporarily>`_. Temporary bootloader parameters are passed to
+the kernel on a single startup. To re-use them you'll need to re-enter them ---
+they don't persist across reboots.
+
+When you're satisfied that a parameter achieves what you want, you can make it
+permanent by `updating GRUB's configuration <#configure-grub-permanently>`_
+then rebooting.
+
+Configure GRUB temporarily
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Reboot, and interrupt GRUB's startup sequence at its bootloader menu.
+
+Highlight the entry --- usually **Ubuntu** --- that would boot Real-time
+Ubuntu.
+
+.. image:: grub-menu.png
+   :alt: GRUB menu
+   :width: 100%
+   :align: center
+
+Press **e** to edit its boot parameters.
+
+Append parameters to the line that starts with ``linux``:
+
+.. image:: grub-edit.png
+   :alt: GRUB edit
+   :width: 100%
+   :align: center
+
+* Leave a space before each new parameter
+
+* Don't add space round ``=`` and other punctuations for each key-value
+
+* Don't add line breaks
+
+Press :kbd:`Ctrl+X` to exit the editor and continue booting with the new
+parameters.
+
+Configure GRUB permanently
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once your device has booted, launch use a text editor to edit ``/etc/default/grub``.
+
+Append parameters to the line that starts with ``GRUB_CMDLINE_LINUX``:
```

Comment:
Hey @farshidtz , sorry, i know you said you're refactoring so this is kinda 'out of scope'.

But wanted to check with you about this line.

If i open the file on my machine there are 2 lines that **start** with `GRUB_CMDLINE_LINUX`.
> GRUB_CMDLINE_LINUX_DEFAULT="quiet splash"
> GRUB_CMDLINE_LINUX="log_buf_len=16M drm.debug=0x1e"

From what I understand (and Googled!) I think it should only be appended to the last line. If that's the case this should probably be worded to be more specific.

E.g. "Append parameters to the line that starts with ``GRUB_CMDLINE_LINUX=``.

Reply:
Both should work, the instructions originally referred to `GRUB_CMDLINE_LINUX_DEFAULT` but I changed it in this PR to not edit the defaults. I guess the whole working is strange because the parameters aren't appended, but rather inserted inside the quotations. 

Changed.